src/main/java/org/cryptomator/cryptofs/CryptoFileSystemModule.java

```diff
@@ -37,6 +40,16 @@ public Cryptor provideCryptor(CryptorProvider cryptorProvider, @PathToVault Path
 		}
 	}
 
+	private String generateFileId(byte[] fileBytes) {
+		try {
+			MessageDigest md = MessageDigest.getInstance("SHA-256");
+			byte[] digest = md.digest(fileBytes);
+			return BaseEncoding.base16().encode(digest, 0, 4);
+		} catch (NoSuchAlgorithmException e) {
+			throw new IllegalStateException("Every Java Platform must support the Message Digest algorithm SHA-256");
```

Comment:
```suggestion
			throw new IllegalStateException("Every Java Platform must support the Message Digest algorithm SHA-256", e);
```

keep the full strack trace

Reply:
put the whole method in an own class and added there the exception in the rethrow